src/ebay-menu-button/menu-button.tsx

```diff
@@ -0,0 +1,176 @@
+    text = '',
+    fixWidth,
+    reverse,
+    expanded: defaultExpanded,
+    noToggleIcon,
+    checked,
+    collapseOnSelect,
+    a11yText,
+    onClick = () => {},
+    onExpand = () => {},
+    onCollapse = () => {},
+    onChange = () => {},
+    onSelect = () => {},
+    children,
+    ...rest
+}) => {
+    const [expanded, setExpanded] = useState(defaultExpanded)
+    const [menuId, setMenuId] = useState<string | undefined>()
+    const buttonRef = useRef(null)
+    const menuRef = useRef()
+
+    const menuItems = filterByType(children, [EbayMenuButtonItem, EbayMenuButtonSeparator])
+
+    const menuButtonLabel = findComponent(children, EbayMenuButtonLabel)
+    const icon = findComponent(children, EbayIcon)
+    const textLabel = text ? <span>{text}</span> : null
+    const label = labelWithIcon(menuButtonLabel || textLabel, icon)
+
+    const wrapperClasses = classnames('menu-button', className, expanded && 'menu-button--expanded')
+
+    const menuClasses = classnames('menu-button__menu', {
+        'menu-button__menu--fix-width': fixWidth,
+        'menu-button__menu--reverse': reverse
+    })
+
+    useEffect(() => {
+        const handleBackgroundClick = (e: React.MouseEvent) => {
+            const menuEl = menuRef.current as HTMLDivElement
+            const menuClicked = menuEl && menuEl.contains(e.target as Node)
+            if (collapseOnSelect || !menuClicked) {
+                setExpanded(false)
+            }
+        }
+
+        if (expanded) {
+            onExpand()
+            // On React 18 useEffect hooks runs synchronous instead of asynchronous as React 17 or prior
+            // causing the event listener to be attached to the document at the same time that the dialog
+            // opens. Adding a timeout so the event is attached after the click event that opened the modal
+            // is finished.
+            setTimeout(() => {
+                document.addEventListener('click', handleBackgroundClick as any, false)
+            })
+        } else if (expanded === false) {
+            onCollapse()
+        }
+        return () => document.removeEventListener('click', handleBackgroundClick as any, false)
+    }, [expanded])
+
+    useEffect(() => {
+        setMenuId(randomId())
+    }, [])
+
+    const handleMenuKeydown = (e) => {
+        handleEscapeKeydown(e, () => {
+            setExpanded(false)
+            buttonRef.current?.focus()
+        })
+    }
+
+    const buttonProps = {
+        ref: buttonRef as any,
+        className: 'menu-button__button',
+        'aria-expanded': !!expanded,
+        'aria-haspopup': true,
+        'aria-label': a11yText,
+        'aria-controls': menuId,
+        onClick: e => {
+            setExpanded(!expanded)
+            onClick(e)
+        },
+        ...rest
+    }
+
+
+    return (
+        <span className={wrapperClasses}>
+            {variant === 'overflow' ?
+                <EbayIconButton icon="overflowHorizontal16" {...buttonProps} /> :
```

Comment:
is this the right one? overflowHorizontal16 used in place of overflow

Reply:
Marko [version](https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-menu-button/index.marko#L87) uses overflowVertical16

Reply:
updated
<img width="616" alt="image" src="https://github.com/eBay/ebayui-core-react/assets/1589482/24046c8e-7154-4e0e-903a-211161af1666">